### PR TITLE
Basic contact page and contact displayer/identifier changes

### PR DESCRIPTION
**Pull request by rowanseymour**
- Switches frontend code to reference contacts by local id rather than backend UUID (which they might not have in some deployments)
- Allows deployments to switch between displaying contact names and anonymous identifiers using setting
- Adds very basic contact read page:

screen shot 2016-05-26 at 08 36 33

### CODE REVIEW

File: casepro/cases/context_processors.py [hunk 1]
```diff
@@ -4,23 +4,6 @@
 import time
 
 from django.conf import settings
-from urlparse import urlparse
-
-
-def contact_ext_url(request):
```
Comment (rowanseymour) [inline casepro/cases/context_processors.py]: Not used anymore - instead external contact URL is based on its own setting

File: casepro/contacts/models.py [hunk 2]
```diff
@@ -245,12 +256,12 @@ def as_json(self, full=True):
         """
         Prepares a contact for JSON serialization
         """
-        result = {'uuid': self.uuid, 'name': self.name}
+        result = {'id': self.pk, 'name': self.get_display_name()}
```
Comment (nicpottier) [inline casepro/contacts/models.py]: Is it still possible for them to see the full backend uuid? Wasn't this something they rolled over before?

Just wondering whether there's value there for them cross referencing with RapidPro

**Comment by nicpottier** [comment]
Looks good apart from that one Q

File: casepro/contacts/models.py (hunk 2 above)
Comment (rowanseymour) [inline casepro/contacts/models.py]: They've only ever been able to see the first 6 chars of the UUID. We added links to the contact in rapidpro on the inbox for admins - with this change those links take a user to the casepro contact page, and from there they can jump to rapidpro by clicking "View External".

**Pull request merged**